Allow configuring shortcuts

Waqar144 opened the pull request:
Fixes #2689

This pull request aims to introduce shortcut configuration to keepassxc.

The design of the respective code is loosely based on KDE's KActionCollection. The ActionCollection manages all actions that can be shortcut configured. These actions are then exposed in the config and a user can assign a different shortcut.

For a first attempt, I have only added actions inside the MainWindow to the ActionCollection. This can be changed / improved upon iteratively

The GUI for configuring actions is pretty simple, it just consists of a `QTreeView` and `QKeySequenceEdit` to change the shortcut.

Note that I have tried to keep the implementation as minimal as possible to make it easy to understand. Things like
an improved gui and ux can always be added on top of an existing feature iteratively.

## Screenshots
![image](https://github.com/keepassxreboot/keepassxc/assets/7696024/eea1751e-1d28-44f8-b2c7-9fa6894dd305)


## Testing strategy
- Add a new shortcut -> trigger that shortcut. Close and reopen Keepassxc, the shortcut should still work with new shortcut
- Clear existing shortcut and restart keepassxc, shortcut should be cleared after restart

## Type of change
- ✅ New feature (change that adds functionality)

droidmonkey commented:
This is spiffy! Thank you for the contribution.

foss- commented:
Wouldn't `Shortcuts` be a sufficiently descriptive naming? After all the item above is named `Security` and not `Configure Security`. Could be `Keyboard Shortcuts` but with the keyboard icon above `Shortcuts` is probably enough?

Waqar144 commented:
> Wouldn't `Shortcuts` be a sufficiently descriptive naming? After all the item above is named `Security` and not `Configure Security`. Could be `Keyboard Shortcuts` but with the keyboard icon above `Shortcuts` is probably enough?

done.

Waqar144 committed:
Allow configuring shortcuts

Closes #2689

This pull request aims to introduce shortcut configuration to keepassxc.

The design of the respective code is loosely based on KDE's KActionCollection. The ActionCollection manages all actions that can be shortcut configured. These actions are then exposed in the config and a user can assign a different shortcut.

Actions inside the MainWindow have been added to the ActionCollection.

The GUI for configuring actions is pretty simple, it just consists of a QTreeView and QKeySequenceEdit to change the shortcut.

Waqar144 committed:
update translations

Waqar144 commented:
@droidmonkey any chance this can get reviewed?

droidmonkey commented:
Sorry this one fell off my radar, will do!

droidmonkey committed:
Code cleanup and enhancements

droidmonkey commented:
I made a ton of improvements to this, ready for merge @phoerious. Most noteably, I am using the ToolTip and not the Name of the action, which lets us make sensible items for the shortcut table.

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/180e71fc-a9b7-43b3-9253-dd8e4fcb617e)

![image](https://github.com/keepassxreboot/keepassxc/assets/2809491/914fa73a-fa90-4a99-905f-f684c4e12c0a)
